altair/vegalite/v5/api.py

```diff
@@ -773,8 +773,13 @@ def condition(predicate, if_true, if_false, **kwargs):
         # dict in the appropriate schema
         if_true = if_true.to_dict()
     elif isinstance(if_true, str):
-        if_true = utils.parse_shorthand(if_true)
-        if_true.update(kwargs)
+        if isinstance(if_false, str):
+            raise ValueError(
+                "A field cannot be used in both the `if_true` and `if_false` branches of a condition. One of the them has to specify a `value` or `datum` definition."
```

Comment:
_`branches of a condition`_ is this the correct way to describe this? I've never heard of this before.

Reply:
I was hesitant about writing it like that too, but went with it because [the VegaLite docs use something similar](https://vega.github.io/vega-lite/docs/condition.html#field)"

> When using a conditional field definition, only value or datum may be specified as the else (outer) branch. 

Do you think changing the error message to the following would be clearer?

> A field cannot be used as both the `if_true` and `if_false` values of a condition. One of the them has to specify a `value` or `datum` definition.

Reply:
Yes, I think that's better (single them).

Reply:
Updated!